### DISCUSSION TITLE
how do I add the hid drivers to the ernal for game pads?

### DISCUSSION THREAD

**Discussion by ulao** [microsoft/WSL#10474] [General]
The Device shows up in lsusb, but not in /dev/input -- well, /dev/input does not exist?

[ 14.812396] usb 1-1: New USB device found, idVendor=16d0, idProduct=0d04, bcdDevice= 2.00
[ 14.812400] usb 1-1: New USB device strings: Mfr=1, Product=2, SerialNumber=4
[ 14.812401] usb 1-1: Product: Gamer-Pro Advanced.2
[ 14.812403] usb 1-1: Manufacturer: Bliss-Box
[ 14.812404] usb 1-1: SerialNumber: 000+
[ 14.825714] input: Bliss-Box Gamer-Pro Advanced.2 as /devices/platform/vhci_hcd.0/usb1/1-1/1-1:1.0/0003:16D0:0D04.0001/input/input0
[ 14.825718] hid-generic 0003:16D0:0D04.0001: input: USB HID v1.11 Gamepad [Bliss-Box Gamer-Pro Advanced.2] on usb-vhci_hcd.0-1/input0

this device uses the common HID type driver. I'd also like to install xinput if not already there.

**Comment by Mikary29** [comment]
You might want to look into  usbipd on this link: https://github.com/dorssel/usbipd-win. And Directions at this link: 
 https://learn.microsoft.com/en-us/windows/wsl/connect-usb. It might help you get closer to what you are trying to accomplish.

**Reply by skunx99** [reply]
That is where this started
https://github.com/dorssel/usbipd-win/issues/700
they led me to wsl for help?
"
Yes, you'll need to add the required drivers to the kernel. That's a WSL issue.
Is your problem resolved now?
"

so I came here looking for help on how to install the missing drivers.

--------------------------------------------------------------------------------------------


Re doing the setup I see a suggested command
sudo update-alternatives --install /usr/local/bin/usbip usbip /usr/lib/linux-tools/*/usbip 100
but it gives me
 priority must be an integer

 dpkg -l | grep linux-tools  says

ii  linux-tools-5.15.0-83                5.15.0-83.92                            amd64        Linux kernel version specific tools for version 5.15.0-83
ii  linux-tools-5.15.0-83-generic        5.15.0-83.92                            amd64        Linux kernel version specific tools for version 5.15.0-83
ii  linux-tools-5.15.0-84                5.15.0-84.93                            amd64        Linux kernel version specific tools for version 5.15.0-84
ii  linux-tools-5.15.0-84-generic        5.15.0-84.93                            amd64        Linux kernel version specific tools for version 5.15.0-84
ii  linux-tools-common                   5.15.0-84.93                            all          Linux kernel version specific tools for version 5.15.0
ii  linux-tools-generic                  5.15.0.84.81                            amd64        Generic Linux kernel tools
ii  linux-tools-virtual                  5.15.0.84.81

can I clean up this mess and just have one version, or is there something I should replace for * above?

**Comment by skunx99** [comment]
That is where this started
https://github.com/dorssel/usbipd-win/issues/700
they led me to wsl for help?
"
Yes, you'll need to add the required drivers to the kernel. That's a WSL issue.
Is your problem resolved now?
"

so I came here looking for help on how to install the missing drivers.

**Comment by skunx99** [comment]
I used purge to remove them all and not specifying a version says to pick from a list. 

Package linux-tools is a virtual package provided by:
  linux-tools-realtime 5.15.0.1032.31
  linux-tools-nvidia-lowlatency-edge 5.19.0.1010.11
  linux-tools-nvidia-lowlatency-5.19 5.19.0.1014.15
  linux-tools-nvidia-5.19 5.19.0.1014.15
  linux-tools-gkeop-5.15 5.15.0.1028.27
  linux-tools-gkeop 5.15.0.1028.27
  linux-tools-virtual-hwe-22.04-edge 6.2.0.33.33~22.04.10
  linux-tools-virtual-hwe-22.04 6.2.0.33.33~22.04.10
  linux-tools-virtual 5.15.0.84.81
  linux-tools-oracle-lts-22.04 5.15.0.1042.37
  linux-tools-oracle-edge 5.19.0.1027.30
  linux-tools-oracle 5.15.0.1040.35
  linux-tools-oem-22.04d 6.5.0.1003.3
  linux-tools-oem-22.04c 6.1.0.1022.22
  linux-tools-oem-22.04b 6.0.0.1021.21
  linux-tools-oem-22.04a 5.17.0.1035.33
  linux-tools-oem-22.04 5.17.0.1035.33
  linux-tools-nvidia-lowlatency 5.15.0.1033.33
  linux-tools-nvidia-hwe-22.04-edge 6.2.0.1010.11
  linux-tools-nvidia-hwe-22.04 6.2.0.1010.11
  linux-tools-nvidia-edge 6.2.0.1003.3~22.04.2
  linux-tools-nvidia-6.2 6.2.0.1010.11
  linux-tools-nvidia 5.15.0.1033.33
  linux-tools-lowlatency-hwe-22.04-edge 6.2.0.1013.13~22.04.10
  linux-tools-lowlatency-hwe-22.04 6.2.0.1013.13~22.04.10
  linux-tools-lowlatency 5.15.0.84.86
  linux-tools-intel-iotg 5.15.0.1039.39
  linux-tools-ibm 5.15.0.1038.34
  linux-tools-gke-5.15 5.15.0.1042.41
  linux-tools-gke 5.15.0.1042.41
  linux-tools-generic-hwe-22.04-edge 6.2.0.33.33~22.04.10
  linux-tools-generic-hwe-22.04 6.2.0.33.33~22.04.10
  linux-tools-generic 5.15.0.84.81
  linux-tools-gcp-lts-22.04 5.15.0.1042.38
  linux-tools-gcp-edge 6.2.0.1014.14~22.04.1
  linux-tools-gcp 6.2.0.1014.14~22.04.1
  linux-tools-aws-lts-22.04 5.15.0.1045.44
  linux-tools-aws-edge 6.2.0.1012.12~22.04.1
  linux-tools-aws 6.2.0.1012.12~22.04.1

not sure what to install. I tried a few but non seem to show the shared usb device from usbipd.
